src/backends/torch/torchlib.cc

```diff
@@ -534,17 +591,25 @@ namespace dd
     try
       {
         inputc.transform(ad);
+
+        _module._native = std::shared_ptr<NativeModule>(
+            NativeFactory::from_template(_template, _template_params, inputc));
+
         if (_module._native)
+          if (!this->_mlmodel._native.empty())
+            torch::load(_module._native, this->_mlmodel._native, _device);
+
+        if (_module._graph)
           {
             auto batchoptional = inputc._dataset.get_batch({ 1 });
             TorchBatch batch = batchoptional.value();
             torch::Tensor td = batch.data[0];
-            _module._native->finalize(td.sizes());
+            _module._graph->finalize(td.sizes());
             // reload params after finalize
             if (!this->_mlmodel._traced.empty())
-              torch::load(_module._native, this->_mlmodel._traced, _device);
-            _module._native->to(_device);
+              torch::load(_module._graph, this->_mlmodel._traced, _device);
           }
+        _module.to(_device);
       }
     catch (...)
       {
```

Comment:
Do we still need the `try { ... } catch (...) { throw; }` ? It seems cumbersome and could be removed with no consequences

Reply:
@beniz this is inspired from the try catch around inputc.transform in caffelib https://github.com/jolibrain/deepdetect/blob/5e982df6b71b55e26715f71f304d534daa59dc7d/src/backends/caffe/caffelib.cc#L1352 , do you have any comments ?

Reply:
`transform` throws by design, if this is what you ask.

Reply:
@beniz What do you mean by that? Is the `try ... catch` here as a remainder for exceptions thrown by `transform`?